src/core/lib/gprpp/ref_counted.h

```diff
@@ -192,6 +192,8 @@ class RefCount {
     return prior == 1;
   }
 
+  bool IsOne() const { return get() == 1; }
```

Comment:
I need to know if I have a unique reference or not.
The two approaches I can see: add an API, or fork this code.
Happy to hear about a third approach, or happy to choose one of the other two.

Reply:
So I think this check ends up being necessary to have an efficient and safe mutation API.

What makes this code safe right now:
- `Slice` does not offer a thread-safe mutation API - if you change a `Slice` instance, it's on you to guarantee that no other threads can access that instance (the normal value type semantics)
- we only perform this check from refcounted slices inside mutating functions from `Slice` instances that own the ref

That we're calling a mutating function makes it illegal to take a ref of this particular `Slice` instance from another thread, and since we prove that there's *only* one ref, there cannot be another instance that could ref this from another thread.

It seems the prudent thing to do is fork `RefCounted` and have a special variant for `Slice` given your API objections.

Reply:
(and done).